flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/IcebergTableSource.java

```diff
@@ -226,4 +229,21 @@ public DynamicTableSource copy() {
   public String asSummaryString() {
     return "Iceberg table source";
   }
+
+  @Override
+  public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext context) {
+    String[] lookupKeys = new String[context.getKeys().length];
+    for (int i = 0; i < lookupKeys.length; i++) {
+      int[] innerKeyArr = context.getKeys()[i];
+      Preconditions.checkArgument(
+          innerKeyArr.length == 1, "Don't support nested lookup keys in iceberg now.");
+      lookupKeys[i] = schema.getFieldNames()[innerKeyArr[0]];
+    }
+
+    TableSchema projectedSchema = getProjectedSchema();
+    loader.open();
```

Comment:
this can be problematic regarding resource management. see more details in PR #6614

Reply:
> this can be problematic regarding resource management. see more details in PR #6614

done

Reply:
done

Reply:
This problem does exist with `tableloader `at present. Therefore, I try to avoid this problem by calling `tableloader.open () `when I load the table. If this problem is to be solved completely, it needs to be solved fundamentally from `tableloader `and I will continue to pay attention to this problem.